### ISSUE TITLE
Define documentation to implement Duolingo for ITD

### ISSUE THREAD

**Issue by kfklein15** [CodeYourFuture/CYF-Staff-Tickets#507]
### Background
We we want to use Duolingo for the English test in ITD and Fundamentals.
For ITD, they have to give us the practice test.

### Acceptance criteria
- Create content for ITD using input of the feedback tickets #504 , #503, #505 , #506 
- Duolingo assessment ITD: must achieve B1 level to pass ITD and be allowed to join Fundamentals
- Approve it
- Update ITD on the Dashboard

### Owner

@kfklein15

### Stakeholder

Trainees and Graduates

### Approver

Sally

**Comment by kfklein15** [comment]
Screenshots have been saved [here](https://drive.google.com/file/d/1jGckz1kr6XikdCelfjtoy4rRMNdkBmRr/view?usp=drive_link).

**Keynotes from KFK**

Must download the Duolingo Desktop app before doing the test. The app has access to read your screen, record screen and microphone.
Duolingo Dekstop app only works for Windows or Mac: https://testcenter.zendesk.com/hc/en-us/articles/5297715242125-What-devices-and-operating-systems-can-I-use-to-take-the-Duolingo-English-Test-
Must have ID/passport in hand
Must be in a silent location by yourself
Test is quite difficult

**Comment by kfklein15** [comment]
**To add to the comms**
- Computer with only Linux doesn't work - has to have either iOS or Windows
- Computer must have a working camera and microphone
- Do not enter the voucher number until you are ready with everything: it is only valid for 20 days. This information is only available **after** the voucher is entered, and there is no possibility to pause.
- Before doing the real test, do this [Practice Test](https://englishtest.duolingo.com/test/practice). This helps you to understand the limited time you have to reply to the questions.

**Comment by kfklein15** [comment]
@SallyMcGrath , could you take a look at the content below and let me know if you are happy with it? 

**Introduction to Digital Content**

**Title**
Take the Duolingo Practice English Test

**Description**
To join the Fundamentals module of the Software Developer course, we need you to have a B1 English level to pass ITD. This is done by you taking the [Duolingo Practice Test](https://englishtest.duolingo.com/test/practice). 

Prep for taking the test:
1. Make sure you have 15-20 minutes available in which you will not be interrupted
1. [Register on the Duolingo website to do the Practice Test ](https://englishtest.duolingo.com/register)
1. Make sure your camera and microphone are working correctly
1. You need to be looking at your camera when doing some of the exercises

Taking the test:
1. Go to the [Duolingo Practice Test](https://englishtest.duolingo.com/test/practice) page
1. Take the test, which will be around 15 minutes
1. Take a screenshot of your result, similar to the example below. 
1. Upload this image to this step.
![Image](https://github.com/CodeYourFuture/CYF-Staff-Tickets/assets/80272258/f0e252d8-1ef2-4b69-9555-4d2c79425ada)

**Important information**
- Please be aware that we expect you to improve your English to B2 level to start the Software Developer Course, so keep practising if you are not on that level yet.
- Understand how the scores of Duolingo work and see a description of what B1 English Level is [on their website](https://englishtest.duolingo.com/scores).

**Comment by SallyMcGrath** [comment]
Oh, it looks like someone deleted the Rainbird doc on English proficiency! What a pain! It had really clear A2 language and good links. And it's linked all over the place, including in messaging.

I have a bunch of edits for this, but before we do that, do you know anything about someone deleting  the document How is proficient in English defined? https://docs.google.com/document/d/1mbNIOiDKhJcdauLIErdAbofueyqy7Q40Ac41n6x3uU0/edit?usp=sharing

**Comment by kfklein15** [comment]
Create a separate ticket to identify where this document used to be linked, so we can update this later - @kfklein15

**Comment by kfklein15** [comment]
Raise ticket for creating a new content: https://github.com/CodeYourFuture/pd-prod-emp/issues/142

**Comment by SallyMcGrath** [comment]
# Introduction to Digital Content


## Take the Duolingo Practice English Test

### Description
To join the Fundamentals  course, you need a B1 English level or higher. B1 is the lowest level we can accept. Take the [Duolingo Practice Test](https://englishtest.duolingo.com/test/practice) to show your level.

#### Prep for taking the test:
1. Find 20 minutes where you will not be interrupted
1. [Register on the Duolingo website to do the Practice Test ](https://englishtest.duolingo.com/register)
1. Test that your camera and microphone are working. You may need to do this test on your phone.
1. You need to look at your camera during some exercises

#### Taking the test:
1. Go to the [Duolingo Practice Test](https://englishtest.duolingo.com/test/practice) page
1. Take the test, which will be around 15 minutes
1. Take a screenshot of your result.
1. Upload your image to this step.
![Image](https://github.com/CodeYourFuture/CYF-Staff-Tickets/assets/80272258/f0e252d8-1ef2-4b69-9555-4d2c79425ada)

#### Important 
- B1 is the lowest possible level we can accept. We can only accept B1 for Fundamentals.
- To join the full Software Developer Course, you must have a B2 or higher level of English
- Read this [description of a B1 English Level](https://englishtest.duolingo.com/scores)

**Comment by kfklein15** [comment]
@SallyMcGrath , could I get your feedback on this by Friday 8th? Thanks

**Comment by kfklein15** [comment]
Asked question about QA testing: https://codeyourfuture.slack.com/archives/C02FX4W0VM4/p1709724593270039

**Comment by kfklein15** [comment]
Upload in QA so far: 

![Image](https://github.com/CodeYourFuture/CYF-Staff-Tickets/assets/80272258/9aca9b80-b0a5-4aaa-823c-8acdefa0e9d3)

**Comment by kfklein15** [comment]
Uploaded in production. 
@SallyMcGrath , I kept some of the previous text (screenshot 2).



![Image](https://github.com/CodeYourFuture/CYF-Staff-Tickets/assets/80272258/eba00dba-1739-4f81-b7a3-e5c4025c8904)



![Image](https://github.com/CodeYourFuture/CYF-Staff-Tickets/assets/80272258/fa791cbd-0cae-404b-beab-368ee60d074e)

**Comment by kfklein15** [comment]
@SallyMcGrath , yesterday, you showed us that your ITD is version 5 (in production). When I created the new Duolingo task, I created it as version 6, since I didn't want to have two things with the same number.
Do you need me to make copies of all steps to version 6 and close the others down? Or should I just use the same version 5, close the other English language task and test it?
Thank you!

**Comment by SallyMcGrath** [comment]
So, even worse I think.

So far as I understand it, what we have to do is, once we are ready to open ITD, find the current version and _paste the new content over it._

This is because to actually start a new version of the course in the dashboard you need a new email? I honestly do not really understand this, but it's what I have written down. If we try to switch to a new version, all 1300 people waiting will be forced to get new emails and we will lose a massive number, so that isn't worth it. We have already done this several times and determined it was an error. 

The downside is that some people who have already done step 1 will have done the old step 1? But this is ok because the old step 1 was Grasshopper and the only reason we changed it is they took down Grasshopper. 

This is my understanding, but like I said, it's on shaky foundations and could be wrong. (I still can't really believe the email thing...)

**Comment by kfklein15** [comment]
Thanks, Sally, yes, you are right!!! Thanks for reminding me about this.

So just to confirm:
1. I will substitute the version 5 previous English with Duolingo
2. Can I close all the previous versions (4 seem to be open still, I can see it on my prod test person) 

If you prefer, we can jump on a call and quickly do this together. But itn's not the end of the world, so we can do it when I am back.

(Can't wait to fix the e-mail problem with the CYF+ people!!!! LOL)

**Comment by SallyMcGrath** [comment]
I think we will have to paste it over the currently open version, whichever one that is. 

We can do this when you get back as we aren't opening ITD before then.

**Comment by kfklein15** [comment]
Updated Staging and Production. 
I did not save it, but ended the date of the previous one and then created a new one.

Screenshot of the environment
![Image](https://github.com/CodeYourFuture/CYF-Staff-Tickets/assets/80272258/c5fd27ec-bc9e-4388-8772-ede0711ce214)

Screenshot of production view for applicants
![Image](https://github.com/CodeYourFuture/CYF-Staff-Tickets/assets/80272258/5a6d85e6-1aeb-480c-8adb-f5baa2fc5cdf)

I also closed all ITD steps from version 3.
